doc/dev/adr/0023-property-support.md

```diff
@@ -0,0 +1,306 @@
+  "@id": "dtmi:propertySketch:PropertySketch;1",
+  "@type": [ "Interface", "Mqtt" ],
+  "payloadFormat": "Json/ecma/404",
+  "propertyTopic": "sample/property/{propertyName}/{action}",
+  "contents": [
+    {
+      "@type": [ "Property", "Fragmented" ],
+      "name": "Baz",
+      "writable": true,
+      "schema": {
+        "@type": "Map",
+        "mapKey": {
+          "name": "propKey",
+          "schema": "string"
+        },
+        "mapValue": {
+          "name": "propValue",
+          "schema": "integer"
+        }
+      }
+    }
+  ]
+}
+```
+
+A dynamically itemized Property placeholder is defined using co-type Fragmented and a schema that is a Map whose contents are dynamically added Properties.
+The Map keys are the names of the dynamic Properties, and the Map values are the Properties themselves.
+
+The Fragmented adjunct type indicates that the Map values have analogous constraints to the Property values of normal static Properties.
+Specifically, Map values in a read response are not nullable, and all values in the Map should be returned on each read Command.
+Map values in a write request are optional and nullable:
+
+* If a key is omitted from the Map in a write request, the value is not written.
+* If a key is included and the value is non-null, the dynamic Property is added if not yet present and updated if already present.
+* If a key is included and the value is null, the dynamic Property is removed from the Map.
+
+If the above-defined error types are used with a dynamically itemized Property, the PropertyValue applies to the Map in aggregate, not to individual entries in the Map.
+
+```json
+{
+  "@context": [ "dtmi:dtdl:context;4", "dtmi:dtdl:extension:mqtt;4" ],
+  "@id": "dtmi:propertySketch:PropertySketch;1",
+  "@type": [ "Interface", "Mqtt" ],
+  "payloadFormat": "Json/ecma/404",
+  "propertyTopic": "sample/property/{propertyName}/{action}",
+  "contents": [
+    {
+      "@type": "Property",
+      "name": "Baz",
+      "writable": true,
+      "schema": {
+        "@type": [ "Object", "PropertyResult" ],
+        "fields": [
+          {
+            "@type": [ "Field", "PropertyValue" ],
+            "name": "baz",
+            "schema": {
+              "@type": "Map",
+              "mapKey": {
+                "name": "propKey",
+                "schema": "string"
+              },
+              "mapValue": {
+                "name": "propValue",
+                "schema": "integer"
+              }
+            }
+          },
+          {
+            "@type": [ "Field", "ReadError", "WriteError" ],
+            "name": "propError",
+            "schema": "dtmi:com:example:BazPropertyError;1"
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+If a modeler wishes to return a Map of error conditions so that each condition can be paired with a dynamic Property in the Map, this must be defined manually.
+In other words, the schema of the Error must be an Object containing a Map field.
```

Comment:
It would be helpful to have some sample codegen output for this scenario in! Not even a full output, but just some expected API surface

Reply:
How about if I just provided an example in DTDL of an error type that included a Map?  Do you think that would be sufficient?

Reply:
Added example DTDL error type definition.

Reply:
I think even just some psuedo code would be helpful!

Reply:
This is super helpful, thank you!